client/src/grpc_client.rs

```diff
@@ -12,35 +12,35 @@ use admission_control_proto::{
 use anyhow::{bail, Result};
 use futures::Future;
 use grpcio::{CallOption, ChannelBuilder, EnvBuilder};
-use libra_crypto::ed25519::*;
 use libra_logger::prelude::*;
+use libra_types::crypto_proxies::EpochInfo;
 use libra_types::{
     access_path::AccessPath,
     account_address::AccountAddress,
     account_config::get_account_resource_or_default,
     account_state_blob::{AccountStateBlob, AccountStateWithProof},
     contract_event::{ContractEvent, EventWithProof},
-    crypto_proxies::ValidatorVerifier,
     get_with_proof::{
         RequestItem, ResponseItem, UpdateToLatestLedgerRequest, UpdateToLatestLedgerResponse,
     },
     transaction::{Transaction, Version},
     vm_error::StatusCode,
 };
 use std::convert::TryFrom;
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
 
 const MAX_GRPC_RETRY_COUNT: u64 = 1;
 
 /// Struct holding dependencies of client.
 pub struct GRPCClient {
```

Comment:
Would be nice to add a comment stating something to the effect of - known_version and epoch_info are expected to update on learning about a new LedgerInfo?